ios/brave-ios/Sources/Brave/Frontend/Share/MenuActivity.swift

```diff
@@ -14,33 +16,98 @@ protocol MenuActivity: UIActivity {
 
 /// A standard activity that will appear in the apps menu and executes a callback when the user selects it
 class BasicMenuActivity: UIActivity, MenuActivity {
+
+  enum ActivityType: String {
```

Comment:
The whole point of `BasicMenuActivity` was to _avoid_ creating specialized types for each activity. If we are going to do this, then we should be extending `UIActivity.ActivityType` instead. e.g.


```swift
extension UIActivity.ActivityType {
  static let copyCleanLink: Self = .init("CopyCleanLink")
  ...
}
```

Then just passing in the activity type into `BasicMenuActivity`.

```swift
init(
    title: String,
    braveSystemImage: String,
    activityType: UIActivity.ActivityType
    callback: @escaping () -> Bool
) { ... }
```

If that's not possible because of the bundle ID requirement, or If you don't want to have to include title/image/activity type in the initializer and want to associate them altogether in one value we can do that too but at that point `ActivityType` should be a `struct`

```swift
struct ActivityType {
  var title: String
  var braveSystemImage: String
  var activityID: String
}
```

Then elsewhere, like BVC

```swift
extension BasicMenuActivity.ActivityType {
  static let cleanCopyLink: Self = .init(...)
  ...
}
```

Then doing

```swift
init(
    activityType: ActivityType,
    callback: @escaping () -> Bool
) { ... }
```

Reply:
Additional changes https://github.com/brave/brave-core/pull/22438/commits/e30a132ec8cdbec932d9b3d3f3b42a65b1115427